frontend/src/components/main-page/groups-tree-page/group-details/group-details.tsx

```diff
@@ -14,7 +28,72 @@ export const GroupDetails: React.FC = () => {
 
   return (
     <React.Fragment>
-      Group Details {JSON.stringify(controller.group)}
+      {controller.group && (
+        <Box
+          sx={{
+            overflowX: 'hidden',
+            overflowY: 'auto',
+            padding: 4,
+            maxWidth: '700px',
+          }}
+        >
+          <Typography variant="h3">Group Information</Typography>
+
+          {Object.keys(controller.group).includes('name') && (
+            <List>
+              <ListItem divider>
+                <ListItemIcon>
+                  <Badge />
+                </ListItemIcon>
+                <ListItemText
+                  primary={(controller.group as ServiceDocsGroup).name}
+                  secondary="Name"
+                />
+              </ListItem>
+            </List>
+          )}
+
+          {Object.keys(controller.group).includes('identifier') && (
```

Comment:
TODO: Think of a better (typesafe) way to do this.

Reply:
I see two options for this. The easiest would be to use the `in` operator:
```
{'name' in controller.group && (
```

However, I am not a huge fan of this, because (at least in the current TS version), it makes refactoring pretty hard: If I hit "Rename Symbol" and rename "group" to "thegroup", then the line with the `in` operator is not automatically refactored.

Thus, my suggested version is a bit more complex, but at least refactorable:


First, we need to extend the Controller like this:

```ts
type GroupWithType = RootGroupWithType | RegularGroupWithType;
interface RootGroupWithType {
  type: 'root-group';
  group: ServiceDocsRootGroup;
}
interface RegularGroupWithType {
  type: 'regular-group';
  group: ServiceDocsGroup;
}
interface Controller {
  groupWithType: GroupWithType | undefined;
}
function useController(): Controller {
  const groupRouterMatch = useMatch(GROUPS_TREE_ROUTES_ABS.group);
  const rootRouterMatch = useMatch(GROUPS_TREE_ROUTES_ABS.root);

  const serviceDocsService = useServiceDocsServiceContext();

  const groupWithType = React.useMemo((): GroupWithType | undefined => {
    if (groupRouterMatch && groupRouterMatch.params.group !== undefined) {
      const theGroup = getGroupByIdentifier(
        groupRouterMatch.params.group,
        serviceDocsService.groupsTree,
      );
      if (!theGroup) {
        return undefined;
      }
      return {
        type: 'regular-group',
        group: theGroup,
      };
    }
    if (rootRouterMatch) {
      return { type: 'root-group', group: serviceDocsService.groupsTree };
    }

    console.warn(
      'Neither the group route nor the root route matched. This should not happen.',
    );
    return undefined;
  }, [groupRouterMatch, rootRouterMatch, serviceDocsService.groupsTree]);

  return {
    groupWithType: groupWithType,
  };
}
```

And now, you can write:

``` ts
{controller.groupWithType.type === 'regular-group' && (
```

Reply:
Awesome, thx :)